### DISCUSSION TITLE
Renovate behavior regarding GoLang "replace" directive

### DISCUSSION THREAD

**Discussion by LeandroEstevez** [renovatebot/renovate#10772] [Request Help]
What does renovate do when there are replace directives in the go.mod file? I have looked through the docs but I have not found anything specific enough.

I did some testing and this is the behavior I saw.

Here we have a dependency (yellow underline) with a version that is not the latest one. And we use a "replace" to replace that version for an even older version.
<img width="734" alt="Screen Shot 2021-07-09 at 3 47 06 PM" src="https://user-images.githubusercontent.com/47323490/125132229-082f3e80-e0d2-11eb-802c-431cc537ff83.png">

Once renovate runs, it will update the dependency inside the "require" statement, and ignore the "replace"
<img width="729" alt="Screen Shot 2021-07-09 at 3 49 47 PM" src="https://user-images.githubusercontent.com/47323490/125132303-2432e000-e0d2-11eb-8d8f-f2f204464f0a.png">

I did some research, I found an option that renovate gives which per the docs it says
postUpdateOptions -> "Enable post-update options to be run after package/artifact updating"

When I tried this, renovate ignored both the "replace" and the dependency on the "require", meaning everything stays the same for both.

I would appreciate it if you guys can give me a detailed explanation of what is the behavior of renovate when working with "replace".

**Comment by secustor** [comment]
Looks like a bug to me. Try replacing    
`replace github.com/aws/aws-sdk-go v1.38.63 => github.com/aws/aws-sdk-go v1.36.24` with 
`replace github.com/aws/aws-sdk-go => github.com/aws/aws-sdk-go v1.36.24`
